contracts/RewardManager.sol

```diff
@@ -145,26 +135,18 @@ contract RewardManager is Ownable, Versionable, Safe {
     emit RewardProgramAdminUpdated(rewardProgramID, newAdmin);
   }
 
-  function addRewardRule(
-    address rewardProgramID,
-    string calldata ruleDID,
-    string calldata tallyRuleDID,
-    string calldata benefitDID
-  ) external onlyHandlers {
-    rule[rewardProgramID][ruleDID] = Rule(tallyRuleDID, benefitDID);
-    emit RewardRuleAdded(rewardProgramID, ruleDID);
-  }
-
-  function removeRewardRule(address rewardProgramID, string calldata ruleDID)
+  function addRewardRule(address rewardProgramID, bytes calldata blob)
     external
     onlyHandlers
   {
-    delete rule[rewardProgramID][ruleDID];
-    emit RewardRuleRemoved(rewardProgramID, ruleDID);
+    rule[rewardProgramID] = keccak256(abi.encodePacked(blob));
```

Comment:
i thought the intent was to store the rule blob on chain, not a hash of it, no?

Reply:
thats actually a good point. @IanCal do you have an opinon on this?